File: SEMESTRE 3/PROG3/ClaseInicial_Chiofalo/src/ParteB/Cliente.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package ParteB;

import java.util.ArrayList;
import java.util.List;

/**
 *
 * @author Emi Chiófalo
 */
public class Cliente {
    
    private int numero; 
    private String razonSocial; 
    private long cuit; 
    private ArrayList <Factura> facturas; 

    public Cliente(int numero, String razonSocial, long cuit) {
        this.numero = numero;
        this.razonSocial = razonSocial;
        this.cuit = cuit;
    }

    public Cliente() {
    }

    public int getNumero() {
        return numero;
    }

    public void setNumero(int numero) {
        this.numero = numero;
    }

    public String getRazonSocial() {
        return razonSocial;
    }

    public void setRazonSocial(String razonSocial) {
        this.razonSocial = razonSocial;
    }

    public long getCuit() {
        return cuit;
    }

    public void setCuit(long cuit) {
        this.cuit = cuit;
    }

    public ArrayList<Factura> getFacturas() {
        return facturas;
    }

    public void addFacturas(Factura factura) {
        if (this.facturas == null) {
            this.facturas = new ArrayList();
        }
        this.facturas.add(factura);
    }
    
    
    
}

File: SEMESTRE 3/LAB3/TP7-TS-ModeloB/Cliente.ts
import { Factura } from "./Factura";

export class Cliente {
    constructor(numero: number, razonSocial: string,
        cuit: number) {
        this._numero = numero;
        this._razonSocial = razonSocial;
        this._cuit = cuit;
    }

    private _numero: number;
    public get numero(): number {
        return this._numero;
    }
    public set numero(value: number) {
        this._numero = value;
    }

    private _razonSocial: string; 
    public get razonSocial(): string {
        return this._razonSocial;
    }
    public set razonSocial(value: string) {
        this._razonSocial = value;
    }

    private _cuit: number; 
    public get cuit(): number {
        return this._cuit;
    }
    public set cuit(value: number) {
        this._cuit = value;
    }

    private _facturas: Array<Factura>;
    public get facturas(): Array<Factura> {
        return this._facturas;
    }
    public set facturas(value: Array<Factura>) {
        this._facturas = value;
    }
    public addFactura(value: Factura) {
        if (this._facturas == null) {
            this._facturas = new Array<Factura>();
        }
        this._facturas.push(value);
    }

    public totalFacturadoXTipoPago(tipoPago: string) : number{
        let total : number = 0;
        this._facturas.forEach(f => {
            if (tipoPago == null) {
                total += f.totalFinal;
            } else if (f.tipoPago == tipoPago){
                total += f.totalFinal;
            }
        });
        return total;
    }

}